marshal/tinyint/unmarshal.go

```diff
@@ -0,0 +1,73 @@
+package tinyint
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+)
+
+func Unmarshal(data []byte, value interface{}) error {
+	switch v := value.(type) {
+	case nil:
+		return nil
+	case *int:
+		return DecInt(data, v)
+	case *uint:
+		return DecUint(data, v)
+	case *int64:
+		return DecInt64(data, v)
+	case *uint64:
+		return DecUint64(data, v)
+	case *int32:
+		return DecInt32(data, v)
+	case *uint32:
+		return DecUint32(data, v)
+	case *int16:
+		return DecInt16(data, v)
+	case *uint16:
+		return DecUint16(data, v)
+	case *int8:
+		return DecInt8(data, v)
+	case *uint8:
+		return DecUint8(data, v)
+	case *big.Int:
+		return DecBigInt(data, v)
+	case *string:
+		return DecString(data, v)
+	case **int:
+		return DecIntR(data, v)
+	case **uint:
+		return DecUintR(data, v)
+	case **int64:
+		return DecInt64R(data, v)
+	case **uint64:
+		return DecUint64R(data, v)
+	case **int32:
+		return DecInt32R(data, v)
+	case **uint32:
+		return DecUint32R(data, v)
+	case **int16:
+		return DecInt16R(data, v)
+	case **uint16:
+		return DecUint16R(data, v)
+	case **int8:
+		return DecInt8R(data, v)
+	case **uint8:
+		return DecUint8R(data, v)
+	case **big.Int:
+		return DecBigIntR(data, v)
+	case **string:
+		return DecStringR(data, v)
+	default:
+		rv := reflect.ValueOf(value)
```

Comment:
Can you please add a comment that describes that you are trying here to address custom type case and you are doing it through reflect because there is no other way, expect generics.

Reply:
Done